pkg/injector/pod_patch.go

```diff
@@ -271,6 +281,59 @@ func profilingEnabled(annotations map[string]string) bool {
 	}
 }
 
+func appendQuantityToResourceList(quantity string, resourceName v1.ResourceName, resourceList v1.ResourceList) (*v1.ResourceList, error) {
+	q, err := resource.ParseQuantity(quantity)
+	if err != nil {
+		return nil, err
+	}
+	resourceList[resourceName] = q
+	return &resourceList, nil
+}
+
+func getResourceRequirements(annotations map[string]string) (*v1.ResourceRequirements, error) {
+	r := v1.ResourceRequirements{
+		Limits:   v1.ResourceList{},
+		Requests: v1.ResourceList{},
+	}
+	cpuLimit, ok := annotations[daprCPULimitKey]
+	if ok {
+		list, err := appendQuantityToResourceList(cpuLimit, v1.ResourceCPU, r.Limits)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing sidecar cpu limit: %s", err)
```

Comment:
I think we can move this err log in the appendQuantityToResourceList method.

Reply:
No, that's a Go anti-pattern. It's too much encapsulation. For example, `appendQuantityToResourceList`  would need to know exactly which type of resource and type (request/limit) is being used, which defeats the purpose of the method.

Reply:
There will be resourceKey to be passed , so type of resource is known and you can still return the error having resourcekey in the error message ..thoughts ?

Reply:
I would rather not. It assigns too much responsibility to the `appendQuantityToResourceList`. Also, the code block for checking the error and returning it would still remain in the calling method, so it doesn't really add anything.